co-authors-plus.php

```diff
@@ -994,24 +994,40 @@ function filter_wp_get_object_terms( $terms, $object_ids, $taxonomies, $args ) {
 	/**
 	 * Filter the count_users_posts() core function to include our correct count
 	 */
-	function filter_count_user_posts( $count, $user_id ) {
+	function filter_count_user_posts( $count = NULL, $user_id, $post_type = NULL ) {
 		$user = get_userdata( $user_id );
 
 		$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
 
 		$term = $this->get_author_term( $user );
 		$guest_term = get_term_by( 'slug', 'cap-' . $user->user_nicename, $this->coauthor_taxonomy );
-		// Only modify the count if it has a linked account with posts or the author exists as a term
-		if ( $user->linked_account && $guest_term->count ) {
-			if ( $term && ! is_wp_error( $term )) {
-				$count = $guest_term->count + $term->count;
-			} else {
-				$count = $guest_term->count;
+
+		if ($post_type == 'post' || empty($post_type)) {
```

Comment:
I am not sure I fully understand this check, also because I don't see `$post_type` being used anywhere in this block? Could you explain what your thoughts are, and why the custom query in the `else` block is needed? :)